server/Dockerfile

```diff
@@ -47,82 +50,62 @@ FROM "${DOCKER_HUB_PROXY}debian:bullseye-slim" as php-build
-    RUN git clone --depth 1 https://github.com/MISP/cti-python-stix2.git; \
-        cd cti-python-stix2 || exit; python3 setup.py bdist_wheel -d /wheels; \
-        echo "-e ." > requirements.txt; pip3 wheel -r requirements.txt --no-cache-dir -w /wheels/
-
-    # Install the new build tool
-    RUN pip3 install build
-
-    # install PyMISP
-    RUN git clone --depth 1 https://github.com/MISP/PyMISP.git; \
-        cd PyMISP || exit; python3 -m build --wheel --outdir /wheels
-
-    # install pydeep2 (drop-in replacement for pydeep)
-    RUN git clone --depth 1 https://github.com/JakubOnderka/pydeep.git; \
-        cd pydeep || exit; python3 setup.py bdist_wheel -d /wheels
-
-    # Grab other modules we need
-    RUN pip3 wheel --no-cache-dir -w /wheels/ plyara pyzmq redis python-magic lief>=0.13.1
-
-    # Remove extra packages due to incompatible requirements.txt files
-    WORKDIR /wheels
-    RUN find . -name "Sphinx*" | tee /dev/stderr | grep -v "Sphinx-1.5.5" | xargs rm -f
-
+    ARG MISP_TAG
+    ARG MISP_COMMIT
+    # Download MISP using git in the /var/www/ directory. Remove unnecessary items.
+    RUN <<-EOF
+            if [ ! -z "${MISP_COMMIT}" ]; then
+                git clone https://github.com/MISP/MISP.git /var/www/MISP && cd /var/www/MISP && git checkout "${MISP_COMMIT}"
+            else
+                git clone --branch "${MISP_TAG}" --depth 1 https://github.com/MISP/MISP.git /var/www/MISP
+            fi
+
+            cd /var/www/MISP || exit; git -c submodule."PyMISP".update=none submodule update --init --recursive .
+
+            # Remove modules only used during MISP build
+            set -- 'coveralls' 'codecov$' 'requests-mock' 'nose'
+            for mod in "$@"; do
+                sed "/{$mod}/d" -i /var/www/MISP/requirements.txt
+            done;
+
+            # Add additional dependencies (container specific)
+            set -- 'redis' 'python-magic' 'lief>=0.13.1' 'pydeep2' 'misp-lib-stix2' 'maec' 'mixbox'
+            for mod in "$@"; do
+                grep -qxF ${mod} /var/www/MISP/requirements.txt || echo ${mod} >> /var/www/MISP/requirements.txt
+            done;
+
+            pip3 wheel --no-cache-dir -w /wheels/ -r /var/www/MISP/requirements.txt
+            
+            # Remove files we do not care for
+            rm -r /var/www/MISP/PyMISP
+            find /var/www/MISP/INSTALL/* ! -name 'MYSQL.sql' -type f -exec rm {} +
+            find /var/www/MISP/INSTALL/* ! -name 'MYSQL.sql' -type l -exec rm {} +
+            rm -rf /var/www/MISP/.git
+
+            # This is PHP related but useful to do in the same step as checkout of MISP
+            mkdir -p /var/www/MISP/app/Vendor && mkdir -p /var/www/MISP/app/Plugin
+EOF
+
+    # Grab module wheels we need
+    RUN pip3 wheel --no-cache-dir -w /wheels/ plyara pyzmq redis python-magic lief>=0.13.1 pymisp pydeep2 misp-lib-stix2 stix cybox maec mixbox
```

Comment:
Agreed. Pinning should be done by MISP. Why do we write (some of) the wheels twice though?

Reply:
I was rushing a bit due to a conflict between vscode quality time and a meeting - apologies. Removed the redundant layer now.